bin/build.sh

```diff
@@ -5,45 +5,153 @@ set -euo pipefail
 cd "$(dirname "${BASH_SOURCE[0]}")/.."
 . bin/stack-helpers.sh
 
-[ $# -eq 1 ] || abort "usage: $(basename "${BASH_SOURCE[0]}") STACK_VERSION"
+STACK_VERSION=${1:-"NAN"}
+REPO=${2:-"heroku/heroku"}
+PUBLISH_SUFFIX=${3:-}
 
-STACK_VERSION=$1
+print_usage(){
+    cat << EOF
+usage: $(basename "${BASH_SOURCE[0]}") STACK_VERSION [IMAGE_REPO] [PUBLISH_SUFFIX]
+
+This script builds heroku base images and writes package lists. It builds
+multi-arch images for heroku-24 and newer, and amd64 images for heroku-22 and
+older. It works in the following scenarios:
+
+- Local builds with Docker Desktop and the \`containerd\` snapshotter. In this
+  mode, all resulting images will be loaded into the local container store.
+  The \`default\` and \`docker-container\` drivers both work in this mode.
+  Note that the \`containerd\` snapshotter is not compatible with \`pack\`.
+
+- For CI tests and package list generation with Docker and the \`default\`
+  Docker driver. In this mode, resulting images will be loaded into the
+  local container store, the package lists generated, but only for amd64.
+  The \`default\` Docker driver is not able to store/retreive multi-arch images
+  locally with the default snapshotter, and the \`containerd\` snapshotter is
+  only available with Docker Desktop. The \`docker-container\` driver will not
+  work in this mode (it can't load any images from the default local store).
+
+- Publishing images in CI with Docker and the \`docker-container\`
+  driver. Pass in a REPO and PUBLISH_SUFFIX argument to publish images
+  directly during the build. Since Docker is unable to store/reference
+  multi-arch images locally, the publish process involves building+pushing
+  an image to a disposable tag, then retagging it. The \`default\` Docker
+  driver will not work in this mode (it can't build cross-architecture).
+EOF
+  exit 1
+}
+
+[[ $STACK_VERSION =~ ^[0-9]+$ ]] || print_usage
+
+docker_container_driver=$(docker buildx inspect | grep -q "docker-container"; echo -n "${PIPESTATUS[1]}")
+
+containerd_snapshotter=$(docker info | grep -q "io.containerd.snapshotter"; echo -n "${PIPESTATUS[1]}")
```

Comment:
I think you can use `docker info -f '{{ .DriverStatus }}' | grep -q -F '[driver-type io.containerd.snapshotter.'` here.